include/SFML/Window/Event.hpp

```diff
@@ -34,195 +34,299 @@
-        SensorEvent sensor; //!< Sensor event parameters (Event::SensorChanged)
+        unsigned int   joystickId{}; //!< Index of the joystick (in range [0 .. Joystick::Count - 1])
+        Joystick::Axis axis{};       //!< Axis on which the joystick moved
+        float          position{};   //!< New position on the axis (in range [-100 .. 100])
     };
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Joystick connection events
+    ///
+    ////////////////////////////////////////////////////////////
+    struct JoystickChanged
+    {
+        unsigned int joystickId{}; //!< Index of the joystick (in range [0 .. Joystick::Count - 1])
+    };
+    struct JoystickConnected : JoystickChanged
+    {
+    };
+    struct JoystickDisconnected : JoystickChanged
+    {
+    };
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Touch events
+    ///
+    ////////////////////////////////////////////////////////////
+    struct TouchChanged
+    {
+        unsigned int finger{}; //!< Index of the finger in case of multi-touch events
+        Vector2i     position; //!< Position of the touch, relative to the top left of the owner window
+    };
+    struct TouchBegan : TouchChanged
+    {
+    };
+    struct TouchMoved : TouchChanged
+    {
+    };
+    struct TouchEnded : TouchChanged
+    {
+    };
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Sensor event
+    ///
+    ////////////////////////////////////////////////////////////
+    struct SensorChanged
+    {
+        Sensor::Type type{}; //!< Type of the sensor
+        Vector3f     value;  //!< Current value of the sensor on the X, Y, and Z axes
+    };
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Default constructor
+    ///
+    /// Sets the event to sf::Event::Empty
+    ///
+    ////////////////////////////////////////////////////////////
+    Event() = default;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Construct from a given sf::Event subtype
+    ///
+    /// \param t Event subtype
+    ///
+    ////////////////////////////////////////////////////////////
+    template <typename T>
+    Event(const T& t);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Check current event type
+    ///
+    /// \return True if template parameter is current event type
+    ///
+    ////////////////////////////////////////////////////////////
+    template <typename T>
+    [[nodiscard]] bool is() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Get particular event type
+    ///
+    /// \return Address of current event type, otherwise nullptr
+    ///
+    ////////////////////////////////////////////////////////////
+    template <typename T>
+    [[nodiscard]] const T* getIf() const;
```

Comment:
I'd prefer `tryGet()`, but there's no regular `get()` (returning a value). So I'd probably just stick to `get()`? But since it returns a pointer (and not some uninitialized value), I'd say this is obvious enough with a simple `get()`. But maybe that's just me.

So essentially one of these:
```cpp
template <typename T>
[[nodiscard]] const T* get() const;
```

```cpp
template <typename T>
[[nodiscard]] const &T get() const;
template <typename T>
[[nodiscard]] const T* tryGet() const;
```

Reply:
Ah, wasn't aware of that. However, it **does** offer a simple [`get()`](https://en.cppreference.com/w/cpp/utility/variant/get) in addition, too.

Reply:
My original and preferred API did include a `get` function and I hope we can add it one day. For now I still think it makes sense to align with the `std::variant` API since users may already be familiar with that.